charts/operator-wandb/charts/weave-trace/templates/deployment.yaml

```diff
@@ -0,0 +1,168 @@
+    {{- end }}
+  annotations:
+    {{- include "wandb.deploymentAnnotations" $ | nindent 4 }}
+    {{- if .Values.deployment.annotations -}}
+    {{-   toYaml .Values.deployment.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "wandb.selectorLabels" $ | nindent 6 }}
+      {{- include "weaveTrace.labels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "wandb.podLabels" . | nindent 8 }}
+        {{- include "weaveTrace.commonLabels" . | nindent 8 }}
+        {{- include "weaveTrace.podLabels" . | nindent 8 }}
+        {{- include "weaveTrace.labels" . | nindent 8 }}
+      annotations:
+        {{- if .Values.pod.annotations -}}
+        {{-   toYaml .Values.pod.annotations | nindent 4 }}
+        {{- end }}
+    spec:
+      {{- if .tolerations }}
+      tolerations:
+        {{- toYaml .tolerations | nindent 8 }}
+      {{- end }}
+      {{- include "wandb.nodeSelector" . | nindent 6 }}
+      {{- include "wandb.priorityClassName" . | nindent 6 }}
+      {{- include "wandb.podSecurityContext" .Values.pod.securityContext | nindent 6 }}
+      {{- if not .Values.migration.useHook }}
+      initContainers:
+        - name: {{ include "weaveTraceMigrate.fullname" . }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          command:
+            - "python"
+            - "migrator.py"
+          env:
+            - name: WF_CLICKHOUSE_HOST
+              value: "{{ .Values.global.clickhouse.host }}"
+            - name: WF_CLICKHOUSE_PORT
+              value: "{{ .Values.global.clickhouse.port }}"
+            - name: WF_CLICKHOUSE_DATABASE
+              value: "{{ .Values.global.clickhouse.database }}"
+            - name: WF_CLICKHOUSE_USER
+              value: "{{ .Values.global.clickhouse.user }}"
+            - name: WF_CLICKHOUSE_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "wandb.clickhouse.passwordSecret" . }}
+                  key: CLICKHOUSE_PASSWORD
+            {{- include "weaveTrace.extraEnv" (dict "global" .Values.global "local" .Values) | nindent 12 }}
+            {{- include "wandb.extraEnvFrom" (dict "root" $ "local" .) | nindent 12 }}
+      {{- end }}
+      containers:
+        - name: {{ include "weaveTrace.fullname" .  }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          env:
+            - name: PORT
+              value: "8080"
+            - name: API_PATH_PREFIX
+              value: "/traces"
+            - name: WANDB_PUBLIC_BASE_URL
+              value: {{ .Values.global.host }}
+            - name: WANDB_BASE_URL
+              value: http://{{ include "weaveTrace.appFullname" . }}.{{ $.Release.Namespace }}.svc.{{ .Values.app.clusterDomain }}:8080/
```

Comment:
Why do we need to set this? Should the service name be enough? We are only setting the service name in other cases so asking for consistency

Reply:
`weave-trace` uses gorilla for auth.  we want to avoid hitting external dns and the additional hop routing through the ALB

Reply:
sorry I guess my question su more ` http://{{ include "weaveTrace.appFullname" . }}:8080` as thats how we do it elsewhere in the code

Reply:
just tested this out, TIL! i will make this change

Reply:
this is done